### PR TITLE
Provide a way to set an initial default input text

### PR DESCRIPTION
**Pull request by irevoire**
Hello, I'm looking on a cli search engine and I was looking for a way to put a default text in the search bar but that was not possible.
Here is an implementation of this feature.

And here is an example of how I want to use it:
https://github.com/irevoire/mieli/commit/ccaae2e13f2b18d93316da795efc735ad9815e08

And here is a video of what it would allow me to do:
[asciicast](https://asciinema.org/a/XSCobUuDGa8MTPHq9fC3cwRKD)

### CODE REVIEW

**Comment by mikaelmello** [comment]
I really like the idea, thanks for that!

Before merging, I'd like to think a bit more about the behavior and the API of this feature.

I'm thinking if I should remove/change the current behavior for default, which uses the default text when the actual input is empty.

Or if I should keep both make their differences clear on the API/documentation

**Comment by irevoire** [comment]
Personally, I would keep both.
But I think we could improve the naming, `text` is not super explicit.

File: src/prompts/text.rs [hunk 1]
```diff
@@ -65,6 +65,11 @@ pub struct Text<'a> {
     /// Message to be presented to the user.
     pub message: &'a str,
 
+    /// The text that will be inputed in the search bar at start.
+    /// If you want to show an example of what can be typed look at [placeholder].
+    /// If you want to set a default value when the user type nothing look at [default].
+    pub text: Option<&'a str>,
```
Comment (mikaelmello) [inline src/prompts/text.rs:71]:
```suggestion
    /// Initial value of the prompt's text input.
    ///
    /// If you want to set a default value for the prompt, returned when the user's submission is empty, see [default].
    pub initial_value: Option<&'a str>,
```

File: src/prompts/text.rs [hunk 2]
```diff
@@ -136,6 +142,12 @@ impl<'a> Text<'a> {
         self
     }
 
+    /// Sets the base input.
```
Comment (mikaelmello) [inline src/prompts/text.rs:145]:
``````suggestion
    /// Sets the initial value of the prompt's text input.
    ///
    /// If you want to set a default value for the prompt, returned when the user's submission is empty, see [with_default].
```

File: src/prompts/text.rs [hunk 3]
```diff
@@ -262,12 +274,11 @@ impl<'a> From<Text<'a>> for TextPrompt<'a> {
...
             help_message: so.help_message,
             formatter: so.formatter,
-            validators: so.validators,
             suggester: so.suggester,
             input: so
                 .placeholder
-                .map(|p| Input::new().with_placeholder(p))
-                .unwrap_or_else(Input::new),
+                .map(|p| Input::new_with(so.text.unwrap_or("")).with_placeholder(p))
+                .unwrap_or_else(|| Input::new_with(so.text.unwrap_or(""))),
```
Comment (mikaelmello) [inline src/prompts/text.rs:281]: Can you think of a way to refactor this 4-line initializer? Looks like a lot to unpack, maybe we could create the `Input` and modify it before calling `Self {}`?

File: src/prompts/text.rs [hunk 4]
```diff
@@ -404,6 +416,9 @@ impl<'a> TextPrompt<'a> {
 
     fn prompt<B: TextBackend>(mut self, backend: &mut B) -> InquireResult<String> {
         let final_answer: String;
+        if !self.input.is_empty() {
+            self.update_suggestions();
+        }
```
Comment (mikaelmello) [inline src/prompts/text.rs:421]: I understand this was necessary to have the correct suggestions loaded before the first render, but it looks a little bit out-of-place, I'll try to think of a way to refactor the Text prompt to process the state before the first render.

File: src/prompts/text.rs (hunk 1 above)
Comment (mikaelmello) [reply src/prompts/text.rs:71]: wdyt of this variable name?

Comment (irevoire) [reply src/prompts/text.rs:71]: Yes, I like it more. 
I think the word `value` in the name is less error-prone :grinning:

File: src/prompts/text.rs (hunk 3 above)
Comment (irevoire) [reply src/prompts/text.rs]: Something like that is enough?
https://github.com/mikaelmello/inquire/pull/34/commits/25ffaa4843e3f2f723df13b70fdeab94fddbc166#diff-2265875200c3d6bf45336f1bd31a5c62580f1511c5e846de5973611255a508cbR274-R280

File: src/prompts/text.rs (hunk 4 above)
Comment (irevoire) [reply src/prompts/text.rs:427]: > I understand this was necessary to have the correct suggestions loaded before the first render
Yes exactly

**Review by mikaelmello — APPROVED** [review]
LGTM! Thanks for the contribution, it's a great idea.

**Pull request merged**